source/server/server.cc

```diff
@@ -592,6 +592,20 @@ void InstanceImpl::initialize(Network::Address::InstanceConstSharedPtr local_add
     }
   }
 
+  // Initialize the regex engine and inject to singleton.
+  const auto& default_regex_engine = bootstrap_.default_regex_engine();
+  if (default_regex_engine.has_typed_config()) {
```

Comment:
bootstrap_.has_default_regex_engine?

where is default engine initialized?

Reply:
> bootstrap_.has_default_regex_engine?

Changed, thanks.

> where is default engine initialized?

It is initialized in a singleton.

https://github.com/envoyproxy/envoy/blob/54e6ecf106e2ee1d8e9078f4299543ca7ba38a2a/source/common/common/regex.cc#L77

Reply:
OK move that initialization up here, so we don't always initialize a GoogleReEngine in global scope?

Reply:
No, since `engine_` is a static variable, a `GoogleReEngine` will be initialized at start up. If user designates an other regex engine or regex engine options, we will re-initialize a new one in bootstrap.

Reply:
Usually, `Regex::Utility::parseRegex` takes the responsibility to parse the regex, and it is a static method. If we do not initialize it with a static variable, we may potentially confront a segfault.
If we do not want a static variable, then I can add a fallback path in `parseRegex` to handle situation without regex engine specific.

Reply:
I'm just asking an `else` block here, you don't need to change everything else and it should still work?

Reply:
Done, with about 10 tests affected and updated.